WordPress/Classes/Services/SiteAssemblyService.swift

```diff
@@ -4,26 +4,184 @@ import Foundation
 // MARK: - EnhancedSiteCreationService
 
 /// Working implementation of a `SiteAssemblyService`.
-final class EnhancedSiteCreationService: SiteAssemblyService {
+///
+final class EnhancedSiteCreationService: LocalCoreDataService, SiteAssemblyService {
 
     // MARK: Properties
 
+    /// A service for interacting with WordPress accounts.
+    private let accountService: AccountService
+
+    /// A service for interacting with WordPress blogs.
+    private let blogService: BlogService
+
+    /// A facade for WPCOM services.
+    private let remoteService: WordPressComServiceRemote
+
+    /// The site creation request that's been enqueued.
+    private var creationRequest: SiteCreationRequest?
+
+    /// This handler is called with changes to the site assembly status.
+    private var statusChangeHandler: SiteAssemblyStatusChangedHandler?
+
+    /// The most recently created blog corresponding to the site creation request; `nil` otherwise.
+    private(set) var createdBlog: Blog?
+
+    // MARK: LocalCoreDataService
+
+    override init(managedObjectContext context: NSManagedObjectContext) {
+        self.accountService = AccountService(managedObjectContext: context)
+        self.blogService = BlogService(managedObjectContext: context)
+
+        let api: WordPressComRestApi
+        if let wpcomApi = accountService.defaultWordPressComAccount()?.wordPressComRestApi {
+            api = wpcomApi
+        } else {
+            api = WordPressComRestApi(userAgent: WPUserAgent.wordPress())
+        }
+        self.remoteService = WordPressComServiceRemote(wordPressComRestApi: api)
+
+        super.init(managedObjectContext: context)
+    }
+
+    // MARK: SiteAssemblyService
+
     private(set) var currentStatus: SiteAssemblyStatus = .idle {
         didSet {
             DispatchQueue.main.async { [weak self] in
-                guard let strongSelf = self else {
+                guard let self = self else {
                     return
                 }
-                strongSelf.statusChangeHandler?(strongSelf.currentStatus)
+                self.statusChangeHandler?(self.currentStatus)
             }
         }
     }
 
-    private(set) var statusChangeHandler: SiteAssemblyStatusChangedHandler?
+    /// This method serves as the entry point for creating an enhanced site. It consists of a few steps:
+    /// 1. The creation request is first validated.
+    /// 1. If a site passes validation, a new service invocation triggers creation.
```

Comment:
Nitpick – these all say step `1.` :)

Reply:
Right you are! Fun fact, Swift documentation supports this for numbered lists, which can be helpful if you need to reorder a sequence, etc.

Here's an example of the documentation generated as-is:
<img width="484" alt="10760_doc" src="https://user-images.githubusercontent.com/221062/50839324-e85a0c80-1314-11e9-8d3c-a196642c07df.png">

Reply:
Right! I think that's the same as standard Markdown – you can number lists however you'll like and they'll be made sequential when you render it.

Reply:
I inferred that you would still like that changed, so I have pushed an additional commit that addresses your feedback.